cmake/ucx-targets.cmake.in

```diff
@@ -7,6 +7,34 @@
 get_filename_component(prefix "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 set(exec_prefix "@exec_prefix@")
 
+set(_ucx_targets_defined "")
+set(_ucx_targets_not_defined "")
+set(_ucx_expected_targets "")
+foreach(_ucx_expected_target IN ITEMS ucx::ucs ucx::uct ucx::ucp)
+  list(APPEND _ucx_expected_targets "${_ucx_expected_target}")
+  if(TARGET "${_ucx_expected_target}")
+    list(APPEND _ucx_targets_defined "${_ucx_expected_target}")
+  else()
+    list(APPEND _ucx_targets_not_defined "${_ucx_expected_target}")
+  endif()
+endforeach()
+unset(_ucx_expected_target)
+if(NOT _ucx_targets_not_defined)
+  unset(_ucx_targets_defined)
+  unset(_ucx_targets_not_defined)
+  unset(_ucx_expected_targets)
+  unset(CMAKE_IMPORT_FILE_VERSION)
+  return()
+endif()
+if(NOT _ucx_targets_defined STREQUAL "")
+  string(REPLACE ";" ", " _ucx_targets_defined_text "${_ucx_targets_defined}")
+  string(REPLACE ";" ", " _ucx_targets_not_defined_text "${_ucx_targets_not_defined}")
+  message(FATAL_ERROR "Some (but not all) ucx targets were already defined.\nTargets Defined: ${_ucx_targets_defined_text}\nTargets not yet defined: ${_ucx_targets_not_defined_text}\n")
```

Comment:
I confirmed that cmake generates same code when producing cmake targets files from libraries, so the guarding approach is right, but since our case is so simple and without any early return, I think we could avoid this generic code and use instead simpler diff below:

```diff
diff --git a/cmake/ucx-targets.cmake.in b/cmake/ucx-targets.cmake.in
index 76642b8..a38daba 100644
--- a/cmake/ucx-targets.cmake.in
+++ b/cmake/ucx-targets.cmake.in
@@ -7,23 +7,29 @@
 get_filename_component(prefix "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 set(exec_prefix "@exec_prefix@")

-add_library(ucx::ucs SHARED IMPORTED)
+if(NOT TARGET ucx::ucs)
+  add_library(ucx::ucs SHARED IMPORTED)

-set_target_properties(ucx::ucs PROPERTIES
-  IMPORTED_LOCATION "@libdir@/libucs.so"
-  INTERFACE_INCLUDE_DIRECTORIES "@includedir@"
-)
+  set_target_properties(ucx::ucs PROPERTIES
+    IMPORTED_LOCATION "@libdir@/libucs.so"
+    INTERFACE_INCLUDE_DIRECTORIES "@includedir@"
+  )
+endif()

-add_library(ucx::ucp SHARED IMPORTED)
+if(NOT TARGET ucx::ucp)
+  add_library(ucx::ucp SHARED IMPORTED)

-set_target_properties(ucx::ucp PROPERTIES
-  IMPORTED_LOCATION "@libdir@/libucp.so"
-  INTERFACE_INCLUDE_DIRECTORIES "@includedir@"
-)
+  set_target_properties(ucx::ucp PROPERTIES
+    IMPORTED_LOCATION "@libdir@/libucp.so"
+    INTERFACE_INCLUDE_DIRECTORIES "@includedir@"
+  )
+endif()

-add_library(ucx::uct SHARED IMPORTED)
+if(NOT TARGET ucx::uct)
+  add_library(ucx::uct SHARED IMPORTED)

-set_target_properties(ucx::uct PROPERTIES
-  IMPORTED_LOCATION "@libdir@/libuct.so"
-  INTERFACE_INCLUDE_DIRECTORIES "@includedir@"
-)
+  set_target_properties(ucx::uct PROPERTIES
+    IMPORTED_LOCATION "@libdir@/libuct.so"
+    INTERFACE_INCLUDE_DIRECTORIES "@includedir@"
+  )
+endif()
```

Reply:
@tvegas1 can you pls create a separate PR with the above suggestion? It looks clean and simple

Reply:
created #9696

Reply:
The simpler fix is fine given that we know all the targets, yes.